lib/nexpose/shared_secret.rb

```diff
@@ -0,0 +1,31 @@
+module Nexpose
+  # SharedSecret class for pairing engines
+  class SharedSecret < APIObject
+    attr_reader :key_string
+    attr_reader :ttl
+
+    def initialize(console, time_to_live)
+      uri = "/data/admin/global/shared-secret?time-to-live=#{time_to_live}"
+      json = JSON.parse(AJAX.put(console, uri))
+      self.from_json(json)
+    end
+
+    def from_json(json)
+      @key_string = json['keyString']
+      @ttl = json['timeToLiveInSeconds']
+    end
+
+    def delete(console)
+      uri = "/data/admin/global/remove-shared-secret?key-string=#{key_string}"
+      AJAX.delete(console, uri)
+    end
+
+    def ==(other)
```

Comment:
Agreed with this. The TTL is metadata. The key string should uniqely identify a shared secret.